uint8[] index # table of output indices which states are requested
---
bool[] state